### ISSUE TITLE
COCKROACH_ARGS environment variable not working as expected

### ISSUE THREAD

**Issue by DennieDev** [cockroachdb/cockroach#122441]
Using COCKROACH_ARGS with flags instead of just a single command results in a failure. 
This is due to a fault in /cockroach/cockroach.sh.

Error:
> ERROR: unknown command "start --insecure" for "cockroach"

Please describe the issue you observed, and any steps we can take to reproduce it:

**To Reproduce**

1. Start a docker container using the COCKROACH_ARGS variable as 'start --insecure'

Example:
```
version: "3.9"
services:
   
  cockroach:
    hostname: "node_{{.Node.Hostname}}"
    image: cockroachdb/cockroach:v23.2.4
    network_mode: host
    environment:
      COCKROACH_ARGS: 'start --insecure'
    deploy:
      replicas: 1
    ports:
      - 26257:26257
      - 26357:26357
      - 18080:8080
```

**Expected behavior**
It should just pass the flags as normal instead of matching it all as a single command, this is due to a fault in the /cockroach/cockroach.sh file. 

**Additional context**
The fix is changing:
```
if [[ -n "$COCKROACH_ARGS" ]]; then
  _main "$COCKROACH_ARGS"
else
  _main "$@"
fi
```
To:
```
if [[ -n "$COCKROACH_ARGS" ]]; then
  _main $COCKROACH_ARGS
else
  _main "$@"
fi
```

Because of the "" it's matching the entire variable as a single command. 

Jira issue: CRDB-37892

**Comment by xinhaoz** [comment]
Tagged @cockroach-dev-inf given [this PR](https://github.com/cockroachdb/cockroach/pull/98899) but feel free to reassign if this is not your area of ownership!

**Comment by WhyAsh5114** [comment]
Hey @rail, I tried to run the latest docker image as a service in a GitHub actions workflow, and still encountered this error:
```
ERROR: unknown command "start-single-node --insecure" for "cockroach"
Failed running "cockroach"
```

Is the PR not yet deployed to the latest image or something?

**Comment by rail** [comment]
It will be available in 24.2
